test/schema_test.cc

```diff
@@ -81,3 +81,443 @@ TEST(SchemaTest, Equality) {
   ASSERT_EQ(schema1, schema5);
   ASSERT_EQ(schema5, schema1);
 }
+
+class BasicShortNameTest : public ::testing::Test {
+ protected:
+  void SetUp() override {
+    field1_ = std::make_unique<iceberg::SchemaField>(1, "Foo", iceberg::int32(), true);
+    field2_ = std::make_unique<iceberg::SchemaField>(2, "Bar", iceberg::string(), true);
+    field3_ = std::make_unique<iceberg::SchemaField>(3, "Foobar", iceberg::int32(), true);
+
+    auto structtype = iceberg::StructType(
```

Comment:
Can we use a shared_ptr for `structtype`? Then we don't need to make an unneccesary  copy it at line 104.